### PR TITLE
[ETH L1 only] Creator Mapping -> All EVMS

### PR DESCRIPTION
**Pull request by MSilb7**
Segment of #3896 where the creator mapping only includes Eth L1

### CODE REVIEW

**Comment by jeff-dude** [comment]
looks like you got past the initial run -- now incremental is broken logic?

**Comment by MSilb7** [comment]
> looks like you got past the initial run -- now incremental is broken logic?

yeah, working through some of them in-between other things. Mostly just column alias collisions.

File: models/contracts/chains/contract_creator_project_mapping/contracts_ethereum_contract_creator_project_mapping.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+ {{
+  config(
+        tags = ['dunesql'],
+        alias = alias('contract_ethereum_creator_project_mapping'),
```
Comment (jeff-dude) [inline models/contracts/chains/contract_creator_project_mapping/contracts_ethereum_contract_creator_project_mapping.sql:4]: file name should mimic schema / alias. with the current file name:
- please add `schema` property above alias, make it `contracts_ethereum`
- update `alias` to `contract_creator_project_mapping`

this will apply univerally

Comment (jeff-dude) [reply models/contracts/chains/contract_creator_project_mapping/contracts_ethereum_contract_creator_project_mapping.sql:4]: we can likely also remove a layer in the directory path, to just be:
`models/contracts/chains/`

since that extra directory layer is just the alias / part of file name

File: models/contracts/chains/contract_creator_project_mapping/contracts_ethereum_contract_creator_project_mapping.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        materialized ='incremental',
+        file_format ='delta',
+        incremental_strategy='merge',
+        unique_key='contract_address',
```
Comment (jeff-dude) [inline models/contracts/chains/contract_creator_project_mapping/contracts_ethereum_contract_creator_project_mapping.sql:8]: unique key should match the test in schema file:
https://github.com/duneanalytics/spellbook/pull/4137/files#diff-6f98b477aba373efb984e25ca4c6291f35b90e24c38090d94db43c38e6debcacR12-R16

please update one or the other to match

File: models/contracts/chains/contract_creator_project_mapping/contracts_ethereum_contract_creator_project_mapping_legacy.sql
Comment (jeff-dude) [inline models/contracts/chains/contract_creator_project_mapping/contracts_ethereum_contract_creator_project_mapping_legacy.sql:1]: you don't have schema set at a project level config. you will need to add schema to all files (which is preferred moving forward anyway). please follow guidelines from above comment

File: models/contracts/chains/find_self_destruct_contracts/contracts_ethereum_find_self_destruct_contracts.sql
Comment (jeff-dude) [inline models/contracts/chains/find_self_destruct_contracts/contracts_ethereum_find_self_destruct_contracts.sql:1]: similar here, we can remove a layer of the directory after `/chains/` as you use that in the alias/file name

File: models/contracts/chains/find_self_destruct_contracts/contracts_find_self_destruct_contracts_schema.yml
Comment (jeff-dude) [inline models/contracts/chains/find_self_destruct_contracts/contracts_find_self_destruct_contracts_schema.yml:1]: once you remove that extra layers in directory path, it makes sense to combine the schema files into one

File: models/contracts/chains/find_self_destruct_contracts/contracts_find_self_destruct_contracts_schema.yml [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+      - &created_block_number
+        name: created_block_number
+        description: "Contract creation block number"
+      - &creation_tx_hash
+        name: creation_tx_hash
+        description: "Contract creation transaction hash"
+      - &contract_address
+        name: contract_address
+        description: "Address of the contract"
+        tests:
```
Comment (jeff-dude) [inline models/contracts/chains/find_self_destruct_contracts/contracts_find_self_destruct_contracts_schema.yml:28]: should have a unique test too

File: models/contracts/contracts_contract_creator_address_list.sql
Comment (jeff-dude) [inline models/contracts/contracts_contract_creator_address_list.sql:1]: this spell already exists under the optimism directory in contracts -- will you remove the old one? which is preferred?

File: models/contracts/contracts_deterministic_contract_creators.sql
Comment (jeff-dude) [inline models/contracts/contracts_deterministic_contract_creators.sql:1]: this one also already exists? what to do with old one?

File: models/contracts/contracts_contract_overrides.sql
Comment (jeff-dude) [inline models/contracts/contracts_contract_overrides.sql:1]: this one also already exists? what to do with old one?

File: models/contracts/contracts_disperse_contracts.sql
Comment (jeff-dude) [inline models/contracts/contracts_disperse_contracts.sql:1]: this one also already exists? what to do with old one?

File: models/contracts/contracts_predeploys.sql
Comment (jeff-dude) [inline models/contracts/contracts_predeploys.sql:1]: this one also already exists? what to do with old one?

File: models/contracts/contracts_predeploys.sql [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+  -- other predeploys
+  select 
+    'optimism' as blockchain
+    ,cast(NULL as varbinary) as trace_creator_address
+    ,cast(NULL as varbinary) as creator_address
+    ,cast(NULL as varbinary) as contract_factory
+    ,contract_address
+    ,'ovm' as contract_project
+    ,contract_name
+    ,from_iso8601_timestamp( '2021-07-06' ) as created_time
```
Comment (jeff-dude) [inline models/contracts/contracts_predeploys.sql:36]: just use `TIMESTAMP '<date>'` here instead

Comment (jeff-dude) [reply models/contracts/contracts_predeploys.sql:36]: applies to all unions here

File: models/contracts/contracts_project_name_mappings.sql
Comment (jeff-dude) [inline models/contracts/contracts_project_name_mappings.sql:1]: this one also already exists? what to do with old one?

File: models/contracts/contracts_system_predeploys.sql
Comment (jeff-dude) [inline models/contracts/contracts_system_predeploys.sql:1]: this one also already exists? what to do with old one?

**Review by jeff-dude — CHANGES_REQUESTED** [review]
left a few comments throughout. mostly organizing the spells. be sure to apply universally on most of the below comments

File: models/contracts/contracts_contract_creator_address_list.sql
Comment (MSilb7) [reply models/contracts/contracts_contract_creator_address_list.sql:1]: Yeah, the plan with all of these is:
- Recreate these files to be "chain agnostic"
- Once we verify that everything works & Optimism is migrated to the new model, we can migrate queries to use the new model.
- Then we can delete the old Optimism models in favor of the cross-chain one.

tl;dr - intending for this to be natively multi-chain, but keeping the old Optimism-specific ones until we get this all working.

**Review by jeff-dude — APPROVED** [review]
lets give it a go
